internal/automanagement/manager.go

```diff
@@ -38,7 +38,10 @@ func (m *Manager) QueryData(ctx context.Context, req *backend.QueryDataRequest)
 func (m *Manager) CheckHealth(ctx context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
 	h, err := m.Get(req.PluginContext)
 	if err != nil {
-		return nil, err
+		return &backend.CheckHealthResult{
+			Status:  backend.HealthStatusError,
+			Message: err.Error(),
```

Comment:
I am a bit worried that error can contain sensitive information, maybe a better approach is automatic logging but not exposing text to the client?

Reply:
The problem is we *need* to return info to the user -- perhaps a special error type?

Reply:
Yeah, I think a special error type can help making intention explicit.

Reply:
I think considering that the user in this case is a grafana admin and is the one setting up the credentials `err.Error()` is probably fine, no?

Reply:
Yeah. Thinking more – errors should not contain sensitive information anyway. If an error consists sensitive info then it's expected to leak to error logs for example anyway. I think most connector libs do not include passwords into error messages.  Seems that this change automatically fixes lot of confusion - and I don't have a better idea in mind. Introducing a special type will require all the plugins to adopt it.

Reply:
In cloud an administrator cannot be trusted as in an on-prem installation. That's the reason why we've added special handling for SQL data sources to not expose any connection information failure to the end user since that could allow an attacker to do "port scans" and such.